monkeytype/compat.py

```diff
@@ -0,0 +1,45 @@
+from typing import Any
```

Comment:
This looks great and I don't think we should change it. But I'm curious why you went with this approach. My first instinct would have been to define a compat "interface" using ABC and then instantiate an instance of that based on which minor version of Python was running. I guess with this approach we always ensure that the implementation details that we rely on are present rather than assuming that they are.

Reply:
I didn't design it this way so much as just evolve it; initially I thought all I needed was a few import shims, then as it became clear I needed alternate implementations of some queries, I just stuck them into the same format. I also thought this could maybe use some cleanup, just ran out of time; that's partly why the PR title says WIP still.

In terms of what we should do, there are a couple orthogonal issues:

1) Define a proper interface and then inherit two versions of it. This seems like probably a clearer way to structure it, regardless of how we decide which version to use.

2) Version-checking vs catching ImportError. In the past I've leaned towards the latter, because it seems like a more direct check of the thing you really care about (and in some cases plays better with someone using an intermediate/development version for whatever reason). But in this case the distinguishing import is pretty subtle (`_GenericAlias`), and I agree it would probably be clearer to version-check.

I'll clean this up.

Reply:
Changed my mind after looking at it a bit, I'm going to leave it as-is. These are all standalone functions, I don't think it's worth the boilerplate to make them all classmethods/staticmethods of some class that's just a namespace. I could make the switch to `sys.version` checking but I'm not sure that matters enough, just going to merge this as-is for now.